### DISCUSSION TITLE
zephyr semtech sx1276 LoRa very poor signal strength

### DISCUSSION THREAD

**Discussion by krupis** [zephyrproject-rtos/zephyr#64127] [Q&A]
Hello. I have recently got into Zephyr and nRF52 microcontrollers. For my first project I have decided to try LoRa point to point communication using the following zephyr sample projects:
https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/drivers/lora/receive
https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/drivers/lora/send

In my board overlay, I added sx1276 property:
```
&spi1 {
	compatible = "nordic,nrf-spi";
	status = "okay";
	cs-gpios = <&gpio0 3 GPIO_ACTIVE_LOW>;
	pinctrl-0 = <&spi1_default>;
	pinctrl-1 = <&spi1_sleep>;
	pinctrl-names = "default", "sleep";
	lora0: sx1276@0 {
		compatible = "semtech,sx1276";
		reg = <0>;
		reset-gpios = <&gpio0 2 GPIO_ACTIVE_LOW>;
		dio-gpios = <&gpio0 28 (GPIO_PULL_DOWN | GPIO_ACTIVE_HIGH)>;
		power-amplifier-output = "pa-boost";
		spi-max-frequency = <125000>;
	};
};

/ {
     aliases {
             lora0 = &lora0;
     };
};
```

For the LoRa modules, I use a simple Ra-02 LoRa module based on semtech sx1278:
https://www.elecrow.com/sx1278-lora-module-433m-10km-ra-02-wireless-spread-spectrum-transmission-socket-for-smart-home-diy.html

For the Microcontroller, I use Adafruit Feather nrf52840
https://learn.adafruit.com/introducing-the-adafruit-nrf52840-feather/downloads

The wiring is as following:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/22237034/ebc72e62-8cda-47d3-9bf4-4afa6265bbe3)

See my github for the full source code:
https://github.com/krupis/nrf52_lora

After programming the modules, I see the following logs:

Sender logs:
```
[00:20:34.811,035] [1B][0m<dbg> sx12xx_common: sx12xx_lora_send: Expected air time of 10 bytes = 289ms[1B][0m
[00:20:35.100,708] [1B][0m<inf> lora_send: Data sent! 
 [1B][0m
[00:20:40.106,384] [1B][0m<dbg> sx12xx_common: sx12xx_lora_send: Expected air time of 10 bytes = 289ms[1B][0m
[00:20:40.396,057] [1B][0m<inf> lora_send: Data sent! 
 [1B][0m
[00:20:45.401,763] [1B][0m<dbg> sx12xx_common: sx12xx_lora_send: Expected air time of 10 bytes = 289ms[1B][0m
[00:20:45.691,406] [1B][0m<inf> lora_send: Data sent! 
 [1B][0m
[00:20:50.697,113] [1B][0m<dbg> sx12xx_common: sx12xx_lora_send: Expected air time of 10 bytes = 289ms[1B][0m
[00:20:50.986,755] [1B][0m<inf> lora_send: Data sent! 
 [1B][0m

```

Receiver logs:
```
[00:20:50.493,743] [1B][0m<inf> lora_send: Received data: helloworld (RSSI:-66dBm, SNR:7dBm)[1B][0m
[00:20:55.789,093] [1B][0m<inf> lora_send: Received data: helloworld (RSSI:-66dBm, SNR:8dBm)[1B][0m
[00:21:01.084,472] [1B][0m<inf> lora_send: Received data: helloworld (RSSI:-66dBm, SNR:7dBm)[1B][0m
[00:21:06.379,791] [1B][0m<inf> lora_send: Received data: helloworld (RSSI:-67dBm, SNR:7dBm)[1B][0m
```

As you can see from the logs above, the RSSI when the both devices are next to each other within 20cm distance is around -67 dBm which is very low...

I have tried placing the devices about 20-30 meters apart and the RSSI: was about -110 dBm.

I Would appreciate any insights. Is that normal? What could be an issue? I am not using perfect antennas (I use whatever came together with the devices), see the picture of the setup below:

![image](https://github.com/zephyrproject-rtos/zephyr/assets/22237034/502b3152-c21f-4e08-908a-fef8f9481bc5)

**Comment by nordicjm** [comment]
The picture tells you all you need to know. You are trying to transmit at 868MHz on a module that says it is for 410-525MHz, this is not going to work. Either change to using a frequency supported by your modules or get other modules that have the correct filters for 868MHz operation

**Comment by krupis** [comment]
> The picture tells you all you need to know. You are trying to transmit at 868MHz on a module that says it is for 410-525MHz, this is not going to work. Either change to using a frequency supported by your modules or get other modules that have the correct filters for 868MHz operation

Can you clarify what do you mean?

Yes you are correct that the module is 433MH. In my code, I have configured the LoRa module for 433Mhz.

I use the following code to initialize LoRa module:
```
bool Initialize_LoRa(bool mode){

	struct lora_modem_config config;
	int ret;

	if (!device_is_ready(lora_dev)) {
		LOG_ERR("LoRa device is not ready\n");
        return 0;
	}

	LOG_INF("LoRa device is ready\n");
	k_sleep(K_MSEC(200));

	config.frequency = 433000000;
	config.bandwidth = BW_125_KHZ;
	config.datarate = SF_10;
	config.preamble_len = 8;
	config.coding_rate = CR_4_5;
	config.iq_inverted = false;
	config.public_network = false;
	config.tx_power = 4;
	config.tx = mode;


	ret = lora_config(lora_dev, &config);
	if (ret < 0) {
		LOG_ERR("LoRa config failed\n");
        return 0;
	}
    return 1;
}
```

Notice:
```
config.frequency = 433000000;
```

I am not fully understanding what is wrong with that

**Reply by nordicjm** [reply]
I missed that you included a link to your github and assumed you were just using the 2 linked samples

**Comment by krupis** [comment]
I have replaced the antennas that came in the package with the 433 mHz antenna from Molex:
https://www.digikey.com/en/products/detail/molex/2042870100/7798377

The results are better by 10 dB, but the results are still quite poor.

I am getting about -50dB to -55dB when the devices are within 10-20cm distance. I would expect the signal strength to be -20 or -30dB at this distance.
